src/backend/EdFi.DataManagementService.Backend.OpenSearch/QueryOpenSearch.cs

```diff
@@ -0,0 +1,127 @@
+    private static JsonArray SortDirective()
+    {
+        return new(new JsonObject { ["_doc"] = new JsonObject { ["order"] = "asc" } });
+    }
+
+    /// <summary>
+    /// Returns OpenSearch index name from the given ResourceInfo.
+    /// OpenSearch indexes are required to be lowercase only, with no periods.
+    /// </summary>
+    private static string IndexFromResourceInfo(ResourceInfo resourceInfo)
+    {
+        return $@"{resourceInfo.ProjectName.Value}${resourceInfo.ResourceName.Value}"
+            .ToLower()
+            .Replace(".", "-");
+    }
+
+    public static async Task<QueryResult> QueryDocuments(
+        IOpenSearchClient client,
+        IQueryRequest queryRequest,
+        ILogger logger
+    )
+    {
+        logger.LogDebug("Entering QueryOpenSearch.Query - {TraceId}", queryRequest.TraceId);
+
+        try
+        {
+            string indexName = IndexFromResourceInfo(queryRequest.ResourceInfo);
+
+            // Build API client requested filters
+            JsonArray terms = [];
+            foreach (var termQuery in queryRequest.TermQueries)
+            {
+                terms.Add(
+                    new JsonObject
+                    {
+                        ["match_phrase"] = new JsonObject
+                        {
+                            [$@"edfidoc.{termQuery.Field}"] = termQuery.Value
+                        }
+                    }
+                );
+            }
+
+            JsonObject query =
+                new()
+                {
+                    ["query"] = new JsonObject { ["bool"] = new JsonObject { ["must"] = terms } },
+                    ["sort"] = SortDirective()
+                };
+
+            if (queryRequest.PaginationParameters.Limit != null)
+            {
+                query.Add(new("size", queryRequest.PaginationParameters.Limit));
+            }
+
+            if (queryRequest.PaginationParameters.Offset != null)
+            {
+                query.Add(new("from", queryRequest.PaginationParameters.Offset));
+            }
+
+            BytesResponse response = await client.Http.PostAsync<BytesResponse>(
+                $"/{indexName}/_search",
+                d => d.Body(query.ToJsonString())
+            );
+
+            JsonNode hits = JsonSerializer.Deserialize<JsonNode>(response.Body)!["hits"]!;
+
+            int totalCount = hits!["total"]!["value"]!.GetValue<int>();
+
+            JsonNode[] documents = hits!["hits"]!
+                .AsArray()
+                // DeepClone() so they can be placed in a new JsonArray
+                .Select(node => node!["_source"]!["edfidoc"]!.DeepClone())!
+                .ToArray()!;
+
+            return new QueryResult.QuerySuccess(new JsonArray(documents), totalCount);
+        }
+        catch (Exception ex)
+        {
+            logger.LogCritical(ex, "Uncaught Query failure - {TraceId}", queryRequest.TraceId);
+            return new QueryResult.UnknownFailure("Unknown Failure");
```

Comment:
Can you confirm if the API client will get an error response that includes the trace id in it?